Assets/Editor/GridMapCreator.cs

```diff
@@ -0,0 +1,379 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class GridMapCreator : EditorWindow
+{
+    #region Private Variables
+
+    private StyleManager styleManager;
+
+    private List<List<Node>> nodes;
+    private List<List<PartsScript>> parts;
+
+    private GUIStyle currentStyle;
+    private GUIStyle empty;
+    private Vector2 nodePos;
+    private Vector2 offset;
+    private Vector2 drag;
+    private Rect MenuBar;
+    private Rect MenuBar2;
+    private GameObject mapGO;
+
+    private bool isErasing;
+    private int buttonWidth = 80;
+    private int mapWidth = 16;
+    private int mapHeight = 9;
+    private int boundsX = 315;
+    private int boundsY = 175;
+
+    #endregion
+
+    #region SetUp
+
+    /// <summary>
+    /// Setting up access to the tool in GUI Window
+    /// </summary>
+    [MenuItem("Window/Level Editor Tool")]
+    private static void OpenWindow()
+    {
+        GridMapCreator window = GetWindow<GridMapCreator>();
+        window.titleContent = new GUIContent("Level Editor Tool");        
+    }
+
+    private void OnEnable()
+    {
+        SetupStyles();
+        SetUpNodesAndParts();
+        SetUpMap();
+    }
+
+    /// <summary>
+    /// Sets up the map game object that will hold all child objects once instantiated. 
+    /// </summary>
+    private void SetUpMap()
+    {
+        try
+        {
+            mapGO = GameObject.FindGameObjectWithTag("Map");
+            RestoreTheMap(mapGO);
+        }
+        catch (Exception e)
+        {
+            throw e;
```

Comment:
This e is unused. I suggest to toast mesage it, the contents of it, so the user will be informed of the error that ocured.